Meta/CMake/Superbuild/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+cmake_minimum_required(VERSION 3.16)
+
+# Variables that we really don't want the user to pass to the serenity cross-build because they will screw things up :^)
+set(LAGOM_ONLY_VARS "CMAKE_CXX_COMPILER;CMAKE_C_COMPILER")
+set(EXCLUDE_FROM_CL_VARS "SERENITY_ARCH;SERENITY_TOOLCHAIN")
+
+# https://stackoverflow.com/a/48555098
+# https://cmake.org/pipermail/cmake/2018-January/067002.html
+# MUST be done before call to 'project'
+get_cmake_property(vars CACHE_VARIABLES)
+foreach(var IN LISTS vars)
+    get_property(currentHelpString CACHE "${var}" PROPERTY HELPSTRING)
+    if("${currentHelpString}" MATCHES "No help, variable specified on the command line." OR "${currentHelpString}" STREQUAL "")
+        message("${var} = [${${var}}]  --  ${currentHelpString}") # uncomment to see the variables being processed
+        if (var IN_LIST EXCLUDE_FROM_CL_VARS)
+            continue()
+        elseif(var IN_LIST LAGOM_ONLY_VARS)
+            list(APPEND LAGOM_CL_ARGS "-D${var}=${${var}}")
+        else()
+            list(APPEND CL_ARGS "-D${var}=${${var}}")
+        endif()
+    endif()
+endforeach()
+
+list(APPEND LAGOM_CL_ARGS ${CL_ARGS})
+
+project(
+    SerenitySuperbuild
+    DESCRIPTION "Orchestrate host and target builds in a single build"
+    LANGUAGES NONE
+)
+
+get_filename_component(
+    SERENITY_SOURCE_DIR "${PROJECT_SOURCE_DIR}/../../.."
+    ABSOLUTE CACHE
+)
+set(SERENITY_ARCH "i686" CACHE STRING "Architecture to build for (i686 or x86_64)")
+set(SERENITY_TOOLCHAIN "GNU" CACHE STRING "Compliler toolchain to use for Serenity (GNU or Clang)")
+set(SERENITY_BUILD_DIR "${PROJECT_BINARY_DIR}/../${SERENITY_ARCH}")
```

Comment:
Going out of the binary directory?

Reply:
This was to make it work exactly the same as the old build, and have the Serenity build in `Build/<arch>`. There's three things this avoids breaking:

1) Meta/serenity.sh, most people's driver script for all things build (except I still see Andreas manually invoking ninja install && ninja image && ninja run in videos sometimes). I have some wip changes on the last commit to update that guy

2) Everyone's IDE settings. Looking around at the documentation for both CLion and VsCode (which should cover 90% or more of active devs) they don't particularly support configuring with a superbuild? So devs will have to manually point their IDE at, for example, the compile_commands.json in the build directory for Serenity. If we don't move that directory, it's one less thing to change in the documentation and a bit less painful to get this in.

3) Ports/. Hoo boy, ports. There's two main kinda of ports, the ones that use CMake where I can just update the path to the Toolchain file, and the ones that are mean and use autotools or some such. Things like prefix path, system include paths, etc are all baked into Ports/.port_include.sh and each individual Ports/port-name/package.sh. the most important ports to not break would probably be gcc, for running a compiler on target, make, CMake, git (?), and all the SDL2-xxx packages that use our fork of SDL2. It'd be nice to not touch those at the same time as trying to split out host tools from the target build.

Reply:
Understandable, would be worth sprinkling a FIXME around this part with a reference to this comment or copy-pasting this into an issue of its own and referencing it.

Reply:
I've added a comment on the tagged line that references this thread.